R/plot.R

```diff
@@ -824,3 +825,345 @@ clusteredOntologyDotplot <- function(ont.res, p.adj=0.05, min.genes=1, cut.h=0.6
 
   enrichplot::dotplot(ont.res, showCategory=top.n, ...)
 }
+
+#' @keywords internal
+estimateCdaSpace <- function(d.counts, d.groups, thresh.pc.var = 0.95, n.dim = 2){
+  cell.loadings <- c()
+  sample.pos <- c()
+  
+  bal <- getRndBalances(d.counts)
+  d.used <- bal$norm
+  
+  if (ncol(d.used) != 2) {
+    for (i in 1:n.dim) {
+      res.remove <- removeGroupEffect(d.used, d.groups, thresh.pc.var = 0.9)
+      cell.loadings <- cbind(cell.loadings, bal$psi %*% res.remove$rotation)
+      sample.pos <- cbind(sample.pos, res.remove$scores)
+      d.used <- d.used - res.remove$used.part
+    }
+  } else {
+    cell.loadings <- bal$psi
+    sample.pos <- d.used
+  }
+  
+  cn <- paste('S', 1:n.dim, sep = '')
+  df.cda <- as.data.frame(sample.pos) %>% set_colnames(cn)
+  df.loadings <- as.data.frame(cell.loadings * 8) %>% set_colnames(cn)
+  return(list(red=df.cda, loadings=df.loadings))
+}
+
+#' @keywords internal
+plotCodaSpaceInner <- function(df.space, df.loadings, d.groups, ref.level, target.level, font.size=3, palette=NULL) {
```

Comment:
An explanatory comment would be useful here.

Reply:
Added